### PR TITLE
Introduce PDF invoices

### PR DESCRIPTION
**Pull request by alexsegura**
Damn simple introduction of PDF invoices based on calculated amounts with taxes. 

Can you test the PDF rendering @Atala ? 
You need to rebuild the PHP container `docker-compose build php`

- [x] Add properties in `LocalBusiness` entity
- [x] Add fields in `RestaurantType`
- [x] Add link to PDF in admin interface / user profile

### CODE REVIEW

File: docker/php/Dockerfile [hunk 1]
```diff
@@ -48,10 +64,7 @@ WORKDIR /srv/coopcycle
 COPY composer.json ./
 COPY composer.lock ./
 
-RUN mkdir -p \
-        var/cache \
-        var/logs \
-        var/sessions \
+RUN mkdir -p var/cache var/logs var/sessions \
```
Comment (Atala) [inline docker/php/Dockerfile]: Remove trailing `\`

File: src/AppBundle/Form/RestaurantType.php [hunk 2]
```diff
@@ -34,27 +34,28 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
-                    'class' => 'AppBundle:Cuisine',
-                    'choice_label' => 'name',
-                    'query_builder' => function (EntityRepository $er) {
-                        return $er->createQueryBuilder('c')->orderBy('c.name', 'ASC');
-                    },
-                ),
-                'allow_add' => true,
-                'allow_delete' => true,
-            ))
+            // // FoodEstablishment
```
Comment (Atala) [inline src/AppBundle/Form/RestaurantType.php]: Why?

File: src/AppBundle/Resources/views/Order/pdf.html.twig [hunk 3]
```diff
@@ -0,0 +1,87 @@
+<!DOCTYPE html>
```
Comment (Atala) [inline src/AppBundle/Resources/views/Order/pdf.html.twig]: I think the filename should be `invoice.html.twig` - it is actually a HTML tempalte.

File: src/AppBundle/Controller/AdminController.php [hunk 4]
```diff
@@ -77,6 +79,24 @@ public function ordersAction(Request $request)
         );
     }
 
+    /**
+     * @Route("/admin/orders/{id}.pdf", name="admin_order_invoice", requirements={"id" = "\d+"})
```
Comment (Atala) [inline src/AppBundle/Controller/AdminController.php]: I think this feature should be accessible to everyone (restaurants, customers too in `/profile`).
Lots of customers will need the pdf for their works invoiving.

File: src/AppBundle/Resources/views/Order/pdf.html.twig [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+          </tbody>
+        </table>
+      </div>
+    </div>
+
+    <div style="clear: both;"></div>
+
+    <h2>{% trans %}Order{% endtrans %}</h2>
+
+    <table style="width: 100%;">
```
Comment (Atala) [inline src/AppBundle/Resources/views/Order/pdf.html.twig]: Should be sthg like `width: 800px;margin: 0 auto;` (nicer)

**Comment by Atala** [comment]
Works for me -> nice job!

capture d ecran 2017-12-06 a 13 42 55

File: src/AppBundle/Controller/AdminController.php (hunk 4 above)
Comment (alexsegura) [reply src/AppBundle/Controller/AdminController.php]: Yep, this is in my TODOLIST

File: docker/php/Dockerfile (hunk 1 above)
Comment (alexsegura) [reply docker/php/Dockerfile]: Why? This will break the multi-line 🤔

File: src/AppBundle/Resources/views/Order/pdf.html.twig (hunk 3 above)
Comment (alexsegura) [reply src/AppBundle/Resources/views/Order/pdf.html.twig]: Yep.

**Comment by alexsegura** [comment]
@Atala I have managed the SIRET number using [additional properties](http://schema.org/additionalProperty), as this is available only in France. 
Right now the implementation is kind of naïve, but at least differences between countries are managed. 

I have faced another problem: restaurant can't actually be saved because the `deliveryService` is `NULL` when using the default one. 
Also, errors display is not consistent when the form is displayed and when it is submitted. 
I'm investigating.

File: docker/php/Dockerfile (hunk 1 above)
Comment (Atala) [reply docker/php/Dockerfile]: Forget it, I didn't look the whole file I thought it was the end of the line -_-

**Comment by Atala** [comment]
Good luck with the Symfony forms! Ping if help needed

**Pull request merged**